cedar-policy-validator/src/types.rs

```diff
@@ -868,14 +907,42 @@ impl IntoIterator for Attributes {
     }
 }
 
+/// Used to tag record types to indicate if their attributes record is open or
+/// closed.
+#[derive(Hash, Ord, PartialOrd, Eq, PartialEq, Debug, Copy, Clone, Serialize)]
+pub enum OpenTag {
+    // The attributes are open. A value of this type may have attributes other
+    // than those listed.
+    OpenAttributes,
+    // The attributes are closed. The attributes for a value of this type must
+    // exactly match the attributes listed in the type.
+    ClosedAttributes,
+}
+
+impl OpenTag {
+    pub(crate) fn is_open(self) -> bool {
+        match self {
+            OpenTag::OpenAttributes => true,
+            OpenTag::ClosedAttributes => false,
+        }
```

Comment:
this could be written `matches!(self, OpenTag::OpenAttributes)`, or even `self == OpenTag::OpenAttributes`.  Not sure if either of those are actually better, just pointing it out.